### DISCUSSION TITLE
How to create a range proof from a message within a signature, like bbs+'s signature?

### DISCUSSION THREAD

**Discussion by ghost** [Consensys/gnark#474] [Q&A]
Here is the process:

1. Sign several messages with bbs+ signature algorithm
2. Generate the range proof from one of the messages within the signature

**Comment by ivokub** [comment]
Hi @sha3n. I'm not completely following the question, but as far as I see then BBS+ signatures are batch-signatures using BLS12-381. Right now we don't support it and we are not planning to add support.

However, if you require only a similar functionality (show that a signature is for a message which is in some range), then you could use EdDSA or ECDSA signatures what we support.
